src/HexagonOptimize.cpp

```diff
@@ -178,7 +180,14 @@ Expr apply_patterns(Expr x, const vector<Pattern> &patterns, const Target &targe
     vector<Expr> matches;
     for (const Pattern &p : patterns) {
 
-        if ((p.flags & (Pattern::v62)) && !target.has_feature(Target::HVX_v62))
+        if ((p.flags & (Pattern::v62orLater)) &&
+            !target.features_any_of({Target::HVX_v62, Target::HVX_v65, Target::HVX_v66}))
```

Comment:
This is minor and I don't see any other issues so I'll leave it up to an optional follow-up PR.